Automattic/jetpack
# JS Components: Add IconTooltip component

This PR adds an IconTooltip component to Jetpack reusable components.

Fixes #25914 

#### Changes proposed in this Pull Request:

* Add `IconTooltip` component reusing [Gutenberg `Popover` component](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/popover)
* Add storybook for `IconTooltip` component
* Ensure the component takes props: `placement`, `animate`, `iconCode`, `title` and `children`
<img width="1187" alt="截圖 2022-09-07 下午4 15 27" src="https://user-images.githubusercontent.com/6869813/188827140-0c683886-441b-4497-983a-57846882d281.png">

#### Other information:


#### Jetpack product discussion


N/A
#### Does this pull request change what data or activity we track or use?


N/A
#### Testing instructions:






* Run `cd projects/js-packages/storybook && npm run storybook:dev` and confirm that you can see `IconTooltip` component in Storybook.
* Under `JS Packages` → `Components` → `IconTooltip`, choose the story `Has Content` and try changing some of the controls. You should be able to adjust the placement, animate, icon code, and tooltip title.
* `Hover` on the icon and you should see the tooltip panel at the set placement.

Note: The tooltip could change the placement automatically when there is no space enough to show the tooltip at the set placement, which is made by the [origin Popover component](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/popover#flip-boolean).
